docs/design/yaml-file.rst

```diff
@@ -0,0 +1,165 @@
+There are many options and features that it doesn't support yet.
+This document will serve as a design document for discuss how to implement the missing features.
+
+Scope
+-----
+
+- Finish the spec to include all the missing options
+- Have consistency around the spec
+- Proper documentation for the end user
+- Allow to specify the spec's version used on the YAML file
+- Show the YAML file on the build process
+- Show/suggest a YAML file at the project creation if it hasn't one
+- Have one source of truth for global configurations 
+
+RTD settings
+------------
+
+No all the RTD settings are applicable to the YAML file,
+others are applicable for each build (or version),
+and others for the global project. 
+
+Not applicable settings
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Those settings can't be on the YAML file because:
+may depend for the initial project setup,
+are planned to be removed,
+security and privacy reasons.
+
+- Project Name
+- Repo URL
+- Repo type
+- Privacy level (this feature is planned to be removed [#privacy-level]_)
+- Project description (this feature is planned to be removed [#project-description]_)
+- Single version (*)
+- Default branch
+- Default version (*)
+- Domains (*)
+- Active versions (*)
+- Translations
+- Subprojects
+- Integrations
+- Notifications
+
+.. note::
+   The items marked with ``(*)`` can be considered to be global settings.
+   But aren't too relevant right now or aren't related to the builds.
+
+Global settings
+~~~~~~~~~~~~~~~
+
+Those settings will be read from the YAML file on the ``default branch``.
+
+- Language
+- Programming Language
+- Project homepage
+- Tags
+- Analytics code
+- Redirects 
+
+Local settings
+~~~~~~~~~~~~~~
+
+Those configurations will be read from the YAML file in the current version that is being built.
+
+Several settings are already implemented and documented on
+https://docs.readthedocs.io/en/latest/yaml-config.html.
+So, they aren't covered with much detail here. 
+
+- Documentation type
+- Project installation (virtual env, requirements file, sphinx configuration file, etc)
+- Additional builds (pdf, epub)
+- Python interpreter
+
+Configuration file
+------------------
+
+Format
+~~~~~~
+
+The file format is based on the YAML spec 1.2 [#yaml-spec]_
+(latest version on the time of this writing).
+
+The file must be on the root directory of the repository, and must be named as:
+
+- ``.readthedocs.yml``
+- ``readthedocs.yml``
+
+Conventions
+~~~~~~~~~~~
+
+The spec of the configuration file must use this conventions.
+
+- Use `[]` to indicate an empty list
+- Use `null` to indicate a null value
+- Use `all` (internal string keyword) to indicate that all options are included on a list with predetermined choices.
+- Use `true`  and `false` as only options on boolean fields
+
+Spec
+~~~~
+
+The current spec is documented on https://docs.readthedocs.io/en/latest/yaml-config.html
+and https://github.com/rtfd/readthedocs-build/blob/master/docs/spec.rst
```

Comment:
I suppose the `spec.rst` file should be completely removed since it's obsolete and a duplication of the `yaml-config.html` (which is way better)

Reply:
Yep, we should remove the old spec.